pymilvus/orm/connections.py

```diff
@@ -272,107 +234,97 @@ def connect(self, alias=Config.MILVUS_CONN_ALIAS, user="", password="", token=""
             >>> from pymilvus import connections
             >>> connections.connect("test", host="localhost", port="19530")
         """
+        # pylint: disable=too-many-statements
 
         def connect_milvus(**kwargs):
-            gh = GrpcHandler(**kwargs)
-
-            t = kwargs.get("timeout")
-            timeout = t if isinstance(t, (int, float)) else Config.MILVUS_CONN_TIMEOUT
-
-            gh._wait_for_channel_ready(timeout=timeout)
-            kwargs.pop('password')
-            kwargs.pop("token", None)
-            kwargs.pop('db_name', None)
-            kwargs.pop('secure', None)
-            kwargs.pop("db_name", "")
-
-            self._connected_alias[alias] = gh
-            self._alias[alias] = copy.deepcopy(kwargs)
-
-        def with_config(config: Tuple) -> bool:
-            for c in config:
-                if c != "":
-                    return True
-
-            return False
+            with self._con_lock:
+                gh = None
+                for key, connection_details in self._alias.items():
+
+                    if (
+                        key in self._connected_alias
+                        and connection_details["address"] == kwargs["address"]
+                        and connection_details["user"] == kwargs["user"]
+                        and connection_details["db_name"] == kwargs["db_name"]
+                        and connection_details["secure"] == kwargs["secure"]
+                    ):
+                        gh = self._connected_alias[key]
+                        break
+
+                if gh is None:
+                    gh = GrpcHandler(**kwargs)
+                    t = kwargs.get("timeout", None)
+                    timeout = t if isinstance(t, (int, float)) else Config.MILVUS_CONN_TIMEOUT
+                    gh._wait_for_channel_ready(timeout=timeout)
+
+                kwargs.pop('password', None)
+                kwargs.pop('token', None)
+
+                self._connected_alias[alias] = gh
```

Comment:
The current action involves overwriting the previous object without reducing the count by one in the _connection_references.